mpl_interactions/generic.py

```diff
@@ -480,3 +483,103 @@ def _onselect(self, verts):
 
     def _ipython_display_(self):
         display(self.fig.canvas)
+
+
+def hyperslicer(
+    arr,
+    cmap=None,
+    norm=None,
+    aspect=None,
+    interpolation=None,
+    alpha=None,
+    vmin=None,
+    vmax=None,
+    origin=None,
+    extent=None,
+    autoscale_cmap=True,
+    filternorm=True,
+    filterrad=4.0,
+    resample=None,
+    url=None,
+    ax=None,
+    slider_format_string=None,
+    title=None,
+    figsize=None,
+    display=True,
+    force_ipywidgets=False,
+    **kwargs,
+):
+
+    params = {}
+    ipympl = notebook_backend()
+    fig, ax = gogogo_figure(ipympl, figsize, ax)
+    use_ipywidgets = ipympl or force_ipywidgets
+    slider_format_strings = create_slider_format_dict(slider_format_string, use_ipywidgets)
+                            
+    name_to_dim = {}
+    slices = [0 for i in range(arr.ndim-2)]
+    #Just pass in an array - no kwargs
+    for i in range(arr.ndim - 2):
+        kwargs[f'axis{i}'] = np.arange(arr.shape[i]) 
+        slider_format_strings[f'axis{i}'] = '{:d}'
+        name_to_dim[f'axis{i}'] = i
+        
+    def update(change, label, key):
+        if label:
+        # continuous
+            params[key] = kwargs[key][change["new"]]
+            label.value = slider_format_strings[key].format(kwargs[key][change["new"]])
+        else:
+        # categorical
+            params[key] = change["new"]
+        if title is not None:
+            ax.set_title(title.format(**params))
+            
+        if arr.ndim>=2:
+            slices[name_to_dim[key]] = change['new']
+            new_data = np.asarray(arr[tuple(slices)])
+            im.set_data(new_data)
+        
+        if autoscale_cmap and (new_data.ndim != 3) and vmin is None and vmax is None:
+            im.norm.autoscale(new_data)
+     
+        if isinstance(vmin, Callable):
+            im.norm.vmin = vmin(**params)
+        if isinstance(vmax, Callable):
+            im.norm.vmax = vmax(**params)
```

Comment:
should these functions also be given the array? Like imagine you had BF and GFP, you'd probably want pretty different scaling for them,. The issue with passing in the array though is that then its inconsistent with the `imshow` function. idk what best

Reply:
I just left these in when I copied interactive imshow. Mostly since i thought eventually having the thresholding features would be nice. would you just have them be `vmin(arr, **params)`?

Reply:
That's what im debating. `vmin(arr, **params)` probably makes more sense here, but it's also confusing that the vmin funcitons for imshow don't take the array.

Reply:
Although maybe what we've learned here is that the vmin functions for imshow should take the array